src/Collections-Abstract/Collection.class.st

```diff
@@ -1542,11 +1542,15 @@ Collection >> reject: aBlock [
 
 { #category : 'enumerating' }
 Collection >> reject: rejectBlock thenCollect: collectBlock [
-	"Utility method to improve readability."
+	"Optimized implementation"
 
 	"(#(1 2 3 4 5) reject: #even thenCollect: [:x|x*10]) >>> #(10 30 50)"
 
-	^ (self reject: rejectBlock) collect: collectBlock
+	| rejectedItems |
+	rejectedItems := self copyEmpty.
+	self do: [ :e | (rejectBlock value: e) ifFalse: [ 
+			rejectedItems add: (collectBlock value: e) ] ].
+	^ rejectedItems
```

Comment:
A small note about the variable name: the method is called "reject", but it still returns items that were **not** rejected, not the items that were rejected as the name would suggest.
So I think it would make more sense to name the variable `selectedItems` as in `select:thenCollect:`.
```suggestion
	| selectedItems |
	selectedItems := self copyEmpty.
	self do: [ :e | (rejectBlock value: e) ifFalse: [ 
			selectedItems add: (collectBlock value: e) ] ].
	^ selectedItems
```